File: codewars/cpp/create_phone_number.cpp
// https://www.codewars.com/kata/525f50e3b73515a6db000b83/solutions/cpp


#include <iostream>

std::string createPhoneNumber(const int arr[10]) {
    std::string area_code, exchange_code, sub_number;

    for(int i = 0; i < 3; ++i) {
        area_code.push_back(arr[i] + '0');
    }

    for(int i = 3; i < 6; ++i) {
        exchange_code.push_back(arr[i] + '0');
    }

    for(int i = 6; i < 10; ++i) {
        sub_number.push_back(arr[i] + '0');
    }

    return "(" + area_code + ") " + exchange_code + "-" + sub_number;
}

int main() {
    int arr[10];

    for(int i = 0; i < 10; ++i) {
        std::cin >> arr[i];
    }

    std::cout << createPhoneNumber(arr) << std::endl;

    return 0;
}

File: codewars/rust/src/create_phone_number.rs
// https://www.codewars.com/kata/525f50e3b73515a6db000b83/train/rust

fn create_phone_number(numbers: &[u8]) -> String {
	let mut phone_number = String::from("(");

	for (i, j) in numbers.iter().enumerate() {
		phone_number.push_str(&j.to_string());
		if i == 2 {
			phone_number.push_str(") ");
		} else if i == 5 {
			phone_number.push_str("-");
		}
	}

	phone_number
}

fn main() {
	println!("{}", create_phone_number(&[1, 2, 3, 4, 5, 6, 7, 8, 9, 9]));
}